File: TPS_Exercicios/tp01/TP01Q01 - Palíndromo em Java/palindromo.java
import java.util.*;

class palindromo {

    public static boolean isFim(String entrada){

        return (entrada.length() == 3 && entrada.charAt(0) == 'F' && entrada.charAt(1) == 'I' && entrada.charAt(2) == 'M');
    }



    public static String verificaPolindromo(String polindromo){

        String resposta = "SIM";
        int esq = 0;
        int dir = polindromo.length() - 1;

        while(esq < dir){
            if(polindromo.charAt(esq) != polindromo.charAt(dir)){
                resposta = "NAO";
            }
            esq++;
            dir--;
        }


        return resposta;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String entrada[] = new String[1000];

        int i = 0;

        do{
            entrada[i] = sc.nextLine();

        }while(isFim(entrada[i++]) == false);

        for(int j = 0; j < i-1; j++){
            System.out.println(verificaPolindromo(entrada[j]));
        }
    }
}

File: TPS_Exercicios/tp01/TP01Q01 - Palíndromo em Java/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <locale.h>


//Verifica se a string analisada é a "FIM"
bool isFim(char entrada[]){
        
    return (strlen(entrada) == 3 && entrada[0] == 'F' && entrada[1] == 'I' && entrada[2] == 'M');
}


void classificarPalindromo(char palavra[]) {
    int esq = 0;
    int dir = strlen(palavra) - 1;
    char resposta[] = "SIM";

    for(int i = 0; i < dir; i++){
        if(!((palavra[i] <= 'Z' && palavra[i] >= 'A') || (palavra[i] <= 'z' && palavra[i] >= 'a') || (palavra[i] < '10' && palavra[i] >= '0'))){
            palavra[i] = 'A';
        }
    }

    while (esq < dir) {
        if (palavra[esq] != palavra[dir]) {
            resposta[0] = 'N';
            resposta[1] = 'A';
            resposta[2] = 'O';
        }
        esq++;
        dir--;
    }

    printf("%s\n", resposta);
}

int main() {    
    
    setlocale(LC_ALL, "pt_BR.UTF-8");  // Linux

    char palavra[500];

    scanf(" %[^\n]", palavra);

    while (isFim(palavra) == false) {

        classificarPalindromo(palavra);
        scanf(" %[^\n]", palavra);
    }

    return 0;
}